rest/handler.go

```diff
@@ -283,6 +278,33 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 		}
 	}
 
+	// Named collections handling
+	if dbc := h.server.GetDbConfig(keyspaceDb); dbc != nil && len(dbc.Scopes) > 0 {
+		// Allow an empty scope to refer to the one SG is running with, rather than falling back to _default
+		if keyspaceScope == nil {
+			keyspaceScope = dbContext.BucketSpec.Scope
+		}
+		if keyspaceCollection == nil {
+			if len(dbc.Scopes[*keyspaceScope].Collections) > 1 {
```

Comment:
You're not doing an existence check here (like you are below with foundScope) - can probably just do this once.